crates/throttle/src/lib.rs

```diff
@@ -2,34 +2,45 @@
 //! The implementation uses an in-memory store, but can be adjusted in the future
 //! to support using a redis-cell equipped redis server to share the throttles
 //! among multiple machines.
-#[cfg(feature = "impl")]
-use mod_redis::{Cmd, FromRedisValue, RedisConnection, RedisError};
-#[cfg(feature = "impl")]
+#[cfg(feature = "redis")]
+use mod_redis::{Cmd, RedisConnection, RedisError};
+
+#[cfg(feature = "redis")]
 use once_cell::sync::OnceCell;
-#[cfg(feature = "impl")]
-use redis_cell_impl::{time, MemoryStore, Rate, RateLimiter, RateQuota};
 use serde::{Deserialize, Serialize};
 use std::convert::TryFrom;
-#[cfg(feature = "impl")]
-use std::sync::Mutex;
+use std::ops::Deref;
 use std::time::Duration;
 use thiserror::Error;
 
-#[cfg(feature = "impl")]
+#[cfg(feature = "redis")]
 pub mod limit;
+#[cfg(feature = "redis")]
+mod throttle;
+
+#[derive(Clone)]
+struct RedisContext {
+    connection: RedisConnection,
+    cl_throttle_command: bool,
```

Comment:
nit: I think either `has_cl_throttle_command` or `cl_throttle_command_available` is slightly clearer. The former is less to type! `has_redis_cell` would also be acceptable and shorter.